src/main/java/com/aws/greengrass/builtin/services/lifecycle/LifecycleIPCEventStreamAgent.java

```diff
@@ -298,4 +325,146 @@ public void discardDeferComponentUpdateFutures() {
+
+                GenericExternalService target;
+                if (component instanceof GenericExternalService) {
+                    target = (GenericExternalService) component;
+                } else {
+                    throw new InvalidArgumentsError("Only external components can be paused.");
+                }
+
+                if (State.RUNNING.equals(target.getState())) {
+                    try {
+                        target.pause();
+                    } catch (ServiceException e) {
+                        throw new ServiceError(String.format("Failed to pause component %s due to : %s",
+                                componentName, e.getMessage()));
+                    }
+                } else {
+                    throw new InvalidArgumentsError(String.format("Component %s is not running", componentName));
+                }
+
+                logger.atDebug().log("Exiting pause request handler");
+                return new PauseComponentResponse();
+            });
+        }
+
+        @Override
+        public void handleStreamEvent(EventStreamJsonMessage streamRequestEvent) {
+
+        }
+    }
+
+    class ResumeComponentHandler extends GeneratedAbstractResumeComponentOperationHandler {
+
+        private final String serviceName;
+
+        protected ResumeComponentHandler(OperationContinuationHandlerContext context) {
+            super(context);
+            serviceName = context.getAuthenticationData().getIdentityLabel();
+        }
+
+        @Override
+        protected void onStreamClosed() {
+
+        }
+
+        @SuppressWarnings("PMD.PreserveStackTrace")
+        @Override
+        public ResumeComponentResponse handleRequest(ResumeComponentRequest request) {
+            logger.atDebug().log("Entering resume request handler");
+            // TODO : Platform check for linux only
+
+            String componentName = request.getComponentName();
+            if (Utils.isEmpty(componentName)) {
+                throw new InvalidArgumentsError("Component name is required.");
+            }
+
+            try {
+                doAuthorization(this.getOperationModelContext().getOperationName(), serviceName,
+                        componentName);
+            } catch (AuthorizationException e) {
+                throw new UnauthorizedError(e.getMessage());
+            }
+
+            GreengrassService component;
+            try {
+                component = kernel.locate(componentName);
+            } catch (ServiceLoadException e) {
+                throw new ResourceNotFoundError();
+            }
+
+            GenericExternalService target;
+            if (component instanceof GenericExternalService) {
+                target = (GenericExternalService) component;
+            } else {
+                throw new InvalidArgumentsError("Only external components can be resumed.");
+            }
+
+            if (target.isPaused()) {
+                try {
+                    target.resume();
```

Comment:
Would resume implementation be synchronous?

Reply:
It would, yes